subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SwiftCompile.java

```diff
@@ -271,7 +271,11 @@ void compile() {
         spec.setModuleName(moduleName.getOrNull());
         spec.setModuleFile(moduleFile.get().getAsFile());
         for (File file : modules.getFiles()) {
-            spec.include(file.getParentFile());
+            if (file.isFile()) {
+                spec.include(file.getParentFile());
+            } else {
+                spec.include(file);
+            }
```

Comment:
Is this because we're passing C++ include directories into this now too?

Reply:
Sort of - it's more an attempt to handle a non-file input in a more reasonable manner.  Basically, we were making the assumption that anything passed as a "module" is a file and if it doesn't happen to be a file, we'll still treat it as if it is a file and get the parent directory.  This just says if it's a file get the parent, if it's a directory, just use the directory.  Another option would just be to ignore anything that's not a file.  Yet another option would be to fail if we get something not a file.  There's probably a question here about how we should support configurability of the include path on a `SwiftCompile` task (if at all).

Reply:
We should be able to once we pull the public header directories out of this (these are currently there to recognize changes to header files).  For now, we get module files, module map files, and public header dirs of cpp dependencies.

Reply:
SGTM